core/src/main/java/hudson/triggers/Trigger.java

```diff
@@ -264,20 +264,24 @@ public void run(AbstractProject p) {
         // Process all triggers, except SCMTriggers when synchronousPolling is set
         for (ParameterizedJobMixIn.ParameterizedJob p : inst.getAllItems(ParameterizedJobMixIn.ParameterizedJob.class)) {
             for (Trigger t : p.getTriggers().values()) {
-                if (! (t instanceof SCMTrigger && scmd.synchronousPolling)) {
-                    LOGGER.log(Level.FINE, "cron checking {0} with spec ‘{1}’", new Object[] {p, t.spec.trim()});
-
-                    if (t.tabs.check(cal)) {
-                        LOGGER.log(Level.CONFIG, "cron triggered {0}", p);
-                        try {
-                            t.run();
-                        } catch (Throwable e) {
-                            // t.run() is a plugin, and some of them throw RuntimeException and other things.
-                            // don't let that cancel the polling activity. report and move on.
-                            LOGGER.log(Level.WARNING, t.getClass().getName() + ".run() failed for " + p, e);
+                if (!(t instanceof SCMTrigger && scmd.synchronousPolling)) {
+                    if (t !=null && t.spec != null && t.tabs != null) {
```

Comment:
tabs can be `null` even when they should not, see the [JIRA issue](https://issues.jenkins-ci.org/browse/JENKINS-36748).  basically an XML pushed with some missing fields, or JobDSL configuration without a cron...
`getTriggers()` should not return `null` - but hey this is about robustness and we have a similar patterns elsewhere in Jenkins for this.